schemas/com.snowplowanalytics.amp/amp_id/jsonschema/1-0-0

```diff
@@ -0,0 +1,30 @@
+{
+  "$schema": "http://iglucentral.com/schemas/com.snowplowanalytics.self-desc/schema/jsonschema/1-0-0#",
+  "self": {
+    "vendor": "com.snowplowanalytics.amp",
+    "name": "amp_id",
+    "format": "jsonschema",
+    "version": "1-0-0"
+  },
+  "description": "Schema for AMP user identification",
+  "properties": {
+    "ampClientId": {
+      "type": "string",
+      "description": "AMP client ID - unique to AMP user",
+      "maxLength": 128
+    },
+    "domainUserId": {
+      "type": [
```

Comment:
Do we really express an optionality of property by `null` and absence of property? We'd like to pick a single approach, e.g. have `type: [string, null]` *and* have it in `required` or have `type: string` and not list in `required`. That's totally okay though if we don't control the data.

Reply:
Ah so maybe this is my schema rustiness combined with a slightly atypical case. We will always have the field, but we have no control over whether or not it contains a value. 

So actually, it seems I should be requiring the field and allowing nulls?

Reply:
Fixed in the rebase now, I believe.

Reply:
> We'd like to pick a single approach, e.g. have `type: [string, null]` *and* have it in `required` or have `type: string` and not list in `required`. 

If a property is not required but its type is simply `type: string`, then sending that property with a `null` value will fail to validate. So for a truly optional field (ie it can be missing, or null), we need *both* `type: [string, null]` and not required.

Reply:
I believe it can't be `null` value unless something goes wrong. But yeah I'll revert to both not required and nullable as this is the safer case.